diff --git a/submodules/discord-rpc/submodules/.clang-format b/submodules/discord-rpc/.clang-format
index 1be8390..e8a5e85 100644
--- a/submodules/discord-rpc/.clang-format
+++ b/submodules/discord-rpc/.clang-format
@@ -34,7 +34,6 @@ DerivePointerAlignment: false
 DisableFormat: false
 FixNamespaceComments: true
 ForEachMacros: []
-IndentCaseLabels: false
 IncludeCategories:
   - Regex: '^("|<)stdafx\.h(pp)?("|>)'
     Priority: -1

